applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+ * @param block  The size of the supporting data or 0 if none
+ */
+typedef void (*amod_response_cb)(struct amod_handle *handle, struct aobj_block *block);
+
+/**
+ * @brief Private structure describing a data in message into the module thread.
+ */
+struct amod_message {
+	/* Audio data block to input */
+	struct aobj_block block;
+
+	/* Sending module's handle */
+	struct amod_handle *tx_handle;
+
+	/* Callback for when the data has been consumed */
+	amod_response_cb response_cb;
+};
+
+/**
+ * @brief A helper structure for describing the modules in a stream as a table.
+ */
+struct amod_table {
+	/* A unique name for the module */
+	char *name;
+
+	/* A unique handle to the module */
+	struct amod_handle *handle;
+
+	/* A unique handle to the module */
+	struct amod_configuration *initial_config;
+
+	/* the modules parameters */
+	struct amod_parameters params;
+
+	/* A unique module context */
+	struct amod_context *context;
+};
+
+/**
+ * @brief  Function for opening a module.
+ *
+ * @param parameters     Pointer to the module set-up parameters
+ * @param configuration  A pointer to the modules configuration (this must also
+ *                       be passed to the #amod_set_configuration() unchanged)
+ * @param name           A unique name for this instance of the module
+ * @param handle         Pointer to the module's private handle
+ * @param context        Pointer to the private context for the module
+ *
+ * @return 0 if successful, error value
+ */
+int amod_open(struct amod_parameters *parameters, struct amod_configuration *configuration,
+	      char *name, struct amod_handle *handle, struct amod_context *context);
+
+/**
+ * @brief  Function to close an open module.
+ *
+ * @param handle  A handle to this module instance
+ *
+ * @return 0 if successful, error value
+ */
+int amod_close(struct amod_handle *handle);
+
+/**
+ * @brief  Function to set the configuration of a module.
+ *
+ * @param handle         A handle to this module instance
+ * @param configuration  A pointer to the modules configuration to set (as
+ *                       passed to the modules #amod_query_resource())
+ *
+ * @return 0 if successful, error value
+ */
+int amod_configuration_set(struct amod_handle *handle, struct amod_configuration *configuration);
+
+/**
+ * @brief  Function to set the configuration of a module.
+ *
+ * @param handle         A handle to this module instance
+ * @param configuration  A pointer to the modules current configuration
+ *
+ * @return 0 if successful, error value
+ */
+int amod_configuration_get(struct amod_handle *handle, struct amod_configuration *configuration);
+
+/**
+ * @brief Function to connect two modules together.
+ *
+ * @param handle_from  The handle for the module for output
+ * @param handle_to    The handle of the module to for input, if the same as ::handle from
+ *                     the data will be put on the ::handle_from output message quere.
```

Comment:
Please revise this documentation, it does not make sense to me at least. How I think it should be, but might have misunderstood:
```suggestion
 * @param handle_to    The handle of the module for input. If it is the same as handle_from
 *                     the data will be put on the handle_from->output_message_queue.
```

Reply:
Yes this is not good. 
What I am trying to say is that if the handles are the same when connection then the input data will be processed in the module and to result sent to this module's output FIFO. Otherwise it would be sent to the next module's input FIFO.

Reply:
Has not been changed